test/common_utils.py

```diff
@@ -187,6 +187,9 @@ def _check_module_exists(name):
 # Enables tests that are slow to run (disabled by default)
 TEST_WITH_SLOW = os.getenv('PYTORCH_TEST_WITH_SLOW', '0') == '1'
 
+# Enables tests for NestedTensors
+TEST_WITH_NESTEDTENSORS = os.getenv('PYTORCH_TEST_WITH_NESTEDTENSORS', '0') == '1'
```

Comment:
I don't see why these tests should only be optionally enabled. They're quick to run, right? We should just turn them on unconditinoally. (Note that named tensor is in a different boat because it is a compile time toggle.)

Reply:
I mostly did this out of caution. The last thing I want is to distribute the hacky dispatch code by default.

Reply:
Are you monkeypatching the actual Tensor class? If the dispatch code all lives in its own module, and doesn't actually affect any other parts of the system, there doesn't seem to be a good reason to have a toggle to disable it; it's already pretty isolated.